muon/domain/messages.js

```diff
@@ -173,18 +171,25 @@ function createMessage(payload, headers, source) {
     if (! headers.event_source) headers.event_source = callingObject();
     if (! headers.channel_op) headers.channel_op = 'normal';
 
+    if (typeof payload === 'object') {
+        payload = JSON.stringify(payload)
+    }
+    
+    if (typeof payload === 'string' || payload instanceof String) {
+        payload = new Buffer(payload)
+    }
+
      var message =  {
        id: uuid.v4(),
        created: new Date().getTime(),
        target_service:  headers.target_service,
        origin_service: headers.origin_service,
-       url: headers.url,
        protocol: headers.protocol,
        step:  headers.step,
        provenance_id: headers.provenance_id,
        content_type: headers.content_type,
        status: headers.status,
-       payload: payload,
+       payload: payload.toJSON().data,
```

Comment:
need to be byte encoded when it arrives on the other side. so it needs to appear in the json as a byte array